instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/ResourceDiscoveryCache.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.resources;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+public class ResourceDiscoveryCache {
+  private static final ConcurrentHashMap<String, Object> cache = new ConcurrentHashMap<>();
```

Comment:
Today, the map seems to contain only one element when it is used in the implementation code. So, should it be a map?

Reply:
I mainly used it because it's a very easy pattern to get concurrency right.

Reply:
removed